### PR TITLE
Fix getValues when shouldUnregister set to false

### PR DESCRIPTION
**Pull request by JeromeDeLeon**
This is the reproducible bug when using `getValues`.
https://codesandbox.io/s/clever-spence-tcdn1

I didn't notice it when merging the PR from before but this problem showed for me when I created a multi-step form and only the last values got displayed on the screen even though I set `shouldUnregister` to `false`.

### CODE REVIEW

File: src/logic/getFieldsValues.ts [hunk 1]
```diff
@@ -14,8 +14,12 @@ export default <TFieldValues extends FieldValues>(
     | { nest: boolean },
 ) => {
   const output = {} as TFieldValues;
+  const fields = {
+    ...fieldsRef.current,
+    ...(unmountFieldsStateRef && unmountFieldsStateRef.current),
```
Comment (bluebill1049) [inline src/logic/getFieldsValues.ts:19]:
```suggestion
    ...unmountFieldsStateRef.current,
```

Comment (bluebill1049) [reply src/logic/getFieldsValues.ts:19]: It defaults to `{}` you don't need conditional there.

Comment (JeromeDeLeon) [reply src/logic/getFieldsValues.ts:19]: The TS is complaining about this one because it may possibly be `undefined`

Comment (bluebill1049) [reply src/logic/getFieldsValues.ts:19]: oh really. let's get it in then! thanks, @JeromeDeLeon.

**Review by bluebill1049 — APPROVED** [review]
👍

File: src/logic/getFieldsValues.ts (hunk 1 above)
Comment (JeromeDeLeon) [reply src/logic/getFieldsValues.ts:19]: Welcome :heart:

**Pull request merged**